### ISSUE TITLE
User understandable directory name for SAF

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#9620]
While trying out the SAF nightlies I have a small remark:

On my test devices the selected SAF directories are listed like
`/tree/primary:cgeo` or
`/tree/37ZUEGHV:test1/testfolder`

I suggest to cut this string and only show everything after `:` (which seems to be the unique separator) before the "real" folder hierarchy starts.

**Comment by eddiemuc** [comment]
@Lineflyer sure can do. Did you open a separate issue so this could be discussed first? Or should I just proceed?

Out of interest and to better understand SAF: I only ever got the /tree/primary Uri format so far on emulator. Is the second one maybe an Uri on an SD card? Or is therr another notable difference between the "real location" behind these two Uris?

**Comment by Lineflyer** [comment]
> @Lineflyer sure can do. Did you open a separate issue so this could be discussed first? Or should I just proceed?

I tend to open dedicated issues for everything I find, so it can be clearly and separately discussed and fixed. 
BTW: We have had to many long threads and discussions about a bunch of topics in one issue lately, so I would suggest to everyone to rather split up issues to have a better structure and "list of work". We can easily use labels or projects to bundle issues.

Regarding "waiting time":
I think the usual process should be to wait a bit (depending on issue complexity) if there are any objections our other solution proposals from other team members. Anyway I think his issue is pretty much clear and you could go for a fix right away. The risk is, that if later a better option comes up, you have double work. ;) ..but unlikely in this case.

> Out of interest and to better understand SAF: I only ever got the /tree/primary Uri format so far on emulator. Is the second one maybe an Uri on an SD card? Or is therr another notable difference between the "real location" behind these two Uris?

On my Google Pixel 3a, Android 11 I always get `/tree/primary:`
On a ZTE device, Android 5.0.1 I always get `/tree/37DA-0DDC:` (sounds like a SD but its the internal root)
Other devices or placing it on SD I did not yet test.

**Comment by eddiemuc** [comment]
Solved in storage_framework branch, looks like this now:

![image](https://user-images.githubusercontent.com/6909759/103307441-50c5bb00-4a10-11eb-83c8-fc87299858ea.png)

But do you think it is ok that there is no hint to the user that this is not a file Uri. User might find this strange.

**Comment by moving-bits** [comment]
How about using some prefix, eg. `[base]/cgeo/maps_old_2`?

**Comment by eddiemuc** [comment]
> How about using some prefix, eg. `[base]/cgeo/maps_old_2`?

What would be the meaning of such a prefix? Does "base" relate to the folder in use, e.g. other values would be offline_map and so on?

Also, you placed a / in front of the path, probably because you assume that the path starts at some sort of root folder. But this is not necessarily the case; in fact I have no idea and found no documentation about how the document Uris are constructed. In case if Media store API I even found URLs where files are only noted using a numeric Id, you can't even tell the file names from them.

**Comment by moving-bits** [comment]
> > How about using some prefix, eg. `[base]/cgeo/maps_old_2`?
> 
> What would be the meaning of such a prefix? Does "base" relate to the folder in use, e.g. other values would be offline_map and so on?

[base] would have a literal meaning, a symbol for the path's origin.

> Also, you placed a / in front of the path, probably because you assume that the path starts at some sort of root folder.

Ok, may be confusing. I just used it as a visual separator, as `[base]cgeo/maps_old_2` looked a bit too compressed for me. Maybe a space will work better in this case.

**Comment by eddiemuc** [comment]
> > > How about using some prefix, eg. `[base]/cgeo/maps_old_2`?
> > 
> > What would be the meaning of such a prefix? Does "base" relate to the folder in use, e.g. other values would be offline_map and so on?
> 
> [base] would have a literal meaning, a symbol for the path's origin.

Ah, but there lies the problem with the document Uris: I can't tell what is the "origin". Or in other words: the "origin" in above case is "/tree/primary" resp "/tree/(complicated number)". We can't display what we don't know.

In general the following info would be available:
* Full uri, e.g. including schema and provider (the above was just the path part of the Uri, and yes the path includes "/tree/primary:"). full uri examples you can find in class PublicLocalStorageTest.
* Whether this is a file uri or a document uri (and at a later point in time maybe whether it's a Media store Uri).
* The folder (base, offline maps, logfiles, ...)
* If it is based on yet another public local folder (e.g. default of offline map is based on based) then we know that folder
* Whether it is the default or a user-defined path

**Comment by fm-sys** [comment]
I would use the `...` e.g. `.../cgeo/maps_old_2` as it's way more understandable for users

**Comment by moving-bits** [comment]
`...` LGTM.

> > [base] would have a literal meaning, a symbol for the path's origin.
> 
> Ah, but there lies the problem with the document Uris: I can't tell what is the "origin". Or in other words: the "origin" in above case is "/tree/primary" resp "/tree/(complicated number)". We can't display what we don't know.

That's why I thought of `[base]` being some universal indicator "that there is some base dir not being displayed", but `...` is even easier to understand, IMHO.

> Whether it is the default or a user-defined path

This might be a useful info, maybe even for the user, but especially for the system info page.

**Comment by eddiemuc** [comment]
Yes, good idea. I will implement `...` Thanks!

Whether dir is user-defined or default is already shown.

**Comment by eddiemuc** [comment]
OK, its implemented and will soon be in storage_framework. Will look like this (same example as above):

![image](https://user-images.githubusercontent.com/6909759/103374627-5a1a5a80-4ad8-11eb-8825-5f7b59710aad.png)

**Comment by eddiemuc** [comment]
Upon request of @fm-sys I replaced the three separate dots with the ellipsis sign in code. In GUI I can't really see the difference butin source it is different (checked into storage_framework).

Can this issue be closed or is there something open from your point of view? Of course fix is only available in storage_framework for now.

**Comment by fm-sys** [comment]
IMHO we can close this. The original issue is solved and if something new comes up, we can create a new issue for it...
